subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/projectmodule/DefaultLocalComponentRegistry.java

```diff
@@ -18,74 +18,33 @@
 
 import org.gradle.api.artifacts.component.BuildIdentifier;
 import org.gradle.api.artifacts.component.ProjectComponentIdentifier;
-import org.gradle.api.internal.project.HoldsProjectState;
-import org.gradle.api.internal.project.ProjectState;
-import org.gradle.api.internal.project.ProjectStateRegistry;
-import org.gradle.api.internal.tasks.NodeExecutionContext;
-import org.gradle.internal.Describables;
 import org.gradle.internal.component.local.model.LocalComponentGraphResolveState;
-import org.gradle.internal.model.CalculatedValueContainer;
-import org.gradle.internal.model.CalculatedValueContainerFactory;
-import org.gradle.internal.model.ValueCalculator;
 
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
-
-public class DefaultLocalComponentRegistry implements LocalComponentRegistry, HoldsProjectState {
+/**
+ * Default implementation of {@link LocalComponentRegistry}.
+ */
+public class DefaultLocalComponentRegistry implements LocalComponentRegistry {
```

Comment:
This is indeed a very thin class. It only exists to contextualize the `BuildTreeLocalComponentProvider`, a build-tree scoped service, with the current build's identifier (as you've noted). 

Given that `BuildTreeLocalComponentProvider` is build-tree scoped, it would have no way of determining the build ID of the caller. That is the point of this class. Otherwise, callers would need to track their build ID. While this is possible, it would require wiring the current build ID across all the users of `LocalComponentRegistry`. I think that would require more work than it is worth.

Reply:
Updated the documentation of this class with some additional info

Reply:
Yeah, there are more callers than I would expect, so I guess it makes sense to keep this.

Your paragraph above would make **excellent** javadoc. 😄